steps/split.py

```diff
@@ -1,37 +1,7 @@
-"""
-This module defines the following routines used by the 'split' step of the regression pipeline:
-
-- ``process_splits``: Defines customizable logic for processing & cleaning the training, validation,
-  and test datasets produced by the data splitting procedure.
-"""
-
 from pandas import DataFrame
 
 
 def process_splits(
     train_df: DataFrame, validation_df: DataFrame, test_df: DataFrame
 ) -> (DataFrame, DataFrame, DataFrame):
-    """
-    Perform additional processing on the split datasets.
-
-    :param train_df: The training dataset produced by the data splitting procedure.
-    :param validation_df: The validation dataset produced by the data splitting procedure.
-    :param test_df: The test dataset produced by the data splitting procedure.
-    :return: A tuple containing, in order: the processed training dataset, the processed
-             validation dataset, and the processed test dataset.
-    """
-
-    def process(df: DataFrame):
-        # Drop invalid data points
-        cleaned = df.dropna()
-        # Filter out invalid fare amounts and trip distance
-        cleaned = cleaned[
-            (cleaned["fare_amount"] > 0)
-            & (cleaned["trip_distance"] < 400)
-            & (cleaned["trip_distance"] > 0)
-            & (cleaned["fare_amount"] < 1000)
-        ]
-
-        return cleaned
```

Comment:
The quality of the model produced in the example will suffer if we don't do this cleaning. I think it's best to preserve this.